app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -6,15 +6,15 @@ def next_setup_path
       auth_method_confirmation_url
     elsif next_setup_choice
       confirmation_path
+    elsif user_session[:mfa_selections]
+      analytics.user_registration_mfa_setup_complete(
+        mfa_method_counts: mfa_context.enabled_two_factor_configuration_counts_hash,
+        enabled_mfa_methods_count: mfa_context.enabled_mfa_methods_count,
+        pii_like_keypaths: [[:mfa_method_counts, :phone]],
+        success: true,
+      )
+      second_mfa_setup_url
```

Comment:
I struggled a bit to find where to interrupt the previous workflow and send the user back on the /second_mfa_setup path. Originally I created a problem where every time a user added a new MFA, including from their account page, it would land on /second_mfa_setup. Whenever this was true: (user_session[:mfa_selections]), the user hadn't finished completion and should still end up at /second_mfa_setup after adding a MFA. I thought that once that became false is when the setup was complete and the user was ready to go on to their intended destination.
I might be totally misunderstanding that method and may need some help with it. It does look like I'm now sending a setup complete event contrary to what's happening.

Reply:
The original workflow is restored